### PR TITLE
add support for packaged libvpx (enabled via TG_OWT_VPX_PACKAGED_BUILD)

### PR DESCRIPTION
**Pull request by vitlav**

### CODE REVIEW

**Comment by john-preston** [comment]
@vitlav Please resolve the conflict and I'll merge.

File: cmake/external.cmake [hunk 1]
```diff
@@ -105,3 +105,25 @@ function(link_libjpeg target_name)
...
+        PRIVATE
+            tg_owt::libvpx
+            tg_owt::libvpx_mmx
+            tg_owt::libvpx_sse2
+            tg_owt::libvpx_ssse3
+            tg_owt::libvpx_sse4
+            tg_owt::libvpx_avx
+            tg_owt::libvpx_avx2
+        )
+    endif()
```
Comment (ilya-fedin) [inline cmake/external.cmake:128]:
```suggestion
    find_package(PkgConfig REQUIRED)
    pkg_check_modules(VPX vpx)
    if (VPX_FOUND)
        target_include_directories(${target_name} PRIVATE ${VPX_INCLUDE_DIRS})
        target_link_libraries(${target_name} PRIVATE ${VPX_LINK_LIBRARIES})
    else()
        include("${CMAKE_CURRENT_LIST_DIR}/cmake/libvpx.cmake")
        target_link_libraries(${target_name}
        PRIVATE
            tg_owt::libvpx
            tg_owt::libvpx_mmx
            tg_owt::libvpx_sse2
            tg_owt::libvpx_ssse3
            tg_owt::libvpx_sse4
            tg_owt::libvpx_avx
            tg_owt::libvpx_avx2
        )
    endif()
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -1749,6 +1742,19 @@ elseif (APPLE)
     )
 endif()
 
+set(libvpx_export)
+if (NOT TG_OWT_VPX_PACKAGED_BUILD)
```
Comment (ilya-fedin) [inline CMakeLists.txt:1746]:
```suggestion
if (NOT VPX_FOUND)
```

Comment (vitlav) [reply CMakeLists.txt]: @ilya-fedin, thanks for the suggestions, I just pushed improved code.

**Comment by vitlav** [comment]
> @vitlav Please resolve the conflict and I'll merge.

@john-preston , done.

**Comment by john-preston** [comment]
I've removed it for now (left in libvpx branch), because after that commit it started failing with bundled libvpx, as if the build results were not added to the final library.

**Comment by vitlav** [comment]
> I've removed it for now (left in libvpx branch), because after that commit it started failing with bundled libvpx, as if the build results were not added to the final library.

I just reproduce it, so I will fix it. Sorry.

**Pull request merged**